source/MaterialXGenShader/SgNode.cpp

```diff
@@ -804,6 +804,47 @@ SgNode* SgNodeGraph::addNode(const Node& node, ShaderGenerator& shadergen)
     return newNode.get();
 }
 
+
+void SgNodeContext::addInputSuffix(SgInput* input, const string& suffix)
```

Comment:
Add to the context the ability to set and remove suffixes for inputs and outputs.